docroot/modules/custom/va_gov_menu_access/src/Service/MenuReductionService.php

```diff
@@ -0,0 +1,595 @@
+   * @param \Drupal\va_gov_user\Service\UserPermsService $user_perms_service
+   *   VaGov user perms service.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManagerInterface $entity_type_manager, AliasManagerInterface $alias_manager, UserPermsService $user_perms_service) {
+    $this->aliasManager = $alias_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->userPermsService = $user_perms_service;
+    $this->settings = $config_factory->get('va_gov_menu_access.settings');
+    $this->buildMenuAccessRules();
+  }
+
+  /**
+   * Reduces a given menu based on user and allowed patterns.
+   *
+   * @param array $form
+   *   The form array by reference.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Instance of FormStateInterface.
+   */
+  public function reduceForm(array &$form, FormStateInterface $form_state) {
+    $this->formState = $form_state;
+    $this->nodeBundle = ($this->formState->getFormObject() instanceof NodeForm) ? $this->formState->getFormObject()->getEntity()->getType() : NULL;
+    $this->currentMenuParent = $this->formState->getValue([
+      'menu',
+      'menu_parent',
+    ]);
+    $this->originalMenuParentOptions = $form['menu']['link']['menu_parent']['#options'];
+    $this->setEmptyMenuParentSelector($form);
+
+    // @todo Remove this nonsense check to activate the remainder of this logic
+    // when the other issues in #2427 have paved the way for adjusting and
+    // testing this logic.
+    if (TRUE) {
+      $this->nuke();
+      return;
+    }
+    // End of @todo removal.  Nothing below this point will run.
+    if ($this->userPermsService->hasAdminRole()) {
+      // User is an admin so no menu reduction needed.
+      $this->nuke();
+      return;
+    }
+
+    $form_id = $this->formState->getFormObject()->getFormId();
+    // RISK: This is a risky pattern tying menu governance to page alias.
+    // Only config joins menu and alias, a config change could break this.
+    // This risk can be mitigated with a different approach once the majority
+    // of the tasks in Issue #2427 have been addressed.
+    $page_alias = !empty($form['path']['widget'][0]['alias']['#default_value']) ? $form['path']['widget'][0]['alias']['#default_value'] : '';
+
+    // Health care region detail pages are special because they have no
+    // pattern based alias and need their own handling.
+    if (($this->nodeBundle !== 'health_care_region_detail_page') || ($this->isLockedPath($page_alias))) {
+      // We have a match, lock it down.
+      $this->disableMenuOptions($form, $form_id);
+    }
+
+    $this->applyVamcMenuRulesForDetailPage($form);
+
+    $this->nuke();
+  }
+
+  /**
+   * Checks if the menu has a parent set.
+   *
+   * @return bool
+   *   TRUE if the menu has a parent, FALSE otherwise.
+   */
+  protected function hasMenuParent() : bool {
+    return (!empty($this->formState->getValue(['menu', 'menu_parent']))) ? TRUE : FALSE;
```

Comment:
```suggestion
    return !empty($this->formState->getValue(['menu', 'menu_parent']));
```

This _should_ work...

Reply:
getValue returns NULL if it finds nothing.  Which then breaks the contract of returning a bool.  Or are you saying it is no longer needed because the return-type declaration of bool will actually cast it to a bool?

Reply:
I had to prove it to myself.  Turns out the return type declaration works for everything but NULL
![image](https://user-images.githubusercontent.com/5752113/119069689-96c9ee00-b9b4-11eb-8d7f-0e92d6f046f1.png)

Reply:
No, but `empty` returns a boolean.

Reply:
if you wanted to cast it `(bool)` is way faster also.